### PR TITLE
Plugin infra

### PR DESCRIPTION
**Pull request by rishabh1403**

### CODE REVIEW

**Comment by rishabh1403** [comment]
@chetansk @pronoy-debnath please review this fast..

File: appServer/index.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+const express = require('express');
+const app = express();
+const glob = require('glob');
+
+/*
+    Loading all the plugins from plugin directory
+*/
+
+glob("./plugins/**/*.routes.js", function (er, files) {
```
Comment (chetansk) [inline appServer/index.js]: Make the plugin folder as a variable, define it at the top and append /**/*.routes.js

File: appServer/index.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    for (let i = 0; i < files.length; i++) {
+        require(files[i])(app);
+    }
+})
+
+/*
+    Starting the server on port 9090.
+    TODO: make the port configurable later
```
Comment (chetansk) [inline appServer/index.js]: You can make the port as env variable now

File: appServer/plugins/template/template.controller.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+*/
+
+
+let getAllUsers = (req, res) => {
+    res.send("All Users");
+}
+
+module.exports = {
+    getAllUsers
+}
```
Comment (chetansk) [inline appServer/plugins/template/template.controller.js]: Add a new line at the end

File: appServer/plugins/template/template.routes.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+/* 
+    Sample plugin entry file. 
+    File must be named as <pluginName>.routes.js and kept in root directory
+    A function is exported which takes in express app instance as argument
```
Comment (chetansk) [inline appServer/plugins/template/template.routes.js]: and kept in *plugin* root directory...

File: appServer/plugins/template/template.routes.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    /*
+        Calling the init function here. {do not edit}
+    */
+    init();
+
+    /* 
+        You can have multiple init functions here with diffferent name.
+        Which can be used to seperate out logic into different groups
+    */
+}
```
Comment (chetansk) [inline appServer/plugins/template/template.routes.js]: add a new line

File: appServer/plugins/three/three.routes.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (chetansk) [inline appServer/plugins/three/three.routes.js]: Do we really need this file ? If not delete

File: appServer/plugins/two/two.routes.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (chetansk) [inline appServer/plugins/two/two.routes.js]: Do we really need this file ? If not delete

**Pull request merged**